### PR TITLE
AutoYaST: add tmpfs support

### PR DESCRIPTION
**Pull request by imobachgs**
## Problem

Replaces #1183 and moves the branch to the `yast/yast-storage-ng` repository.

In SLE 12 (with old *yast2-storage*) AutoYaST was able to import/export tmpfs file systems. Such a feature is missing in SLE 15.

* https://jira.suse.com/browse/SLE-11308
* PBI: https://trello.com/c/xsRpwgTB/2195-5-autoyast-support-for-tmpfs

## Solution

tmpfs file systems are defined using a `<drive>` section which type is `CT_TMPFS`. Each `<partition>` section in the drive corresponds to a tmpfs.

```xml
<drive>
  <type config:type="symbol">CT_TMPFS</type>
  <partitions config:type="list">
    <partition>
      <mount>/srv</mount>
      <fstopt>size=128M</fstopt>
    </partition>
    <partition>
      <mount>/var/tmp</mount>
      <fstopt>size=2G</fstopt>
    </partition>
  </partitions>
</drive>
```

If you prefer, you can use one drive section per tmpfs.~~Actually, AutoYaST will use this approach when cloning a system.~~

```xml
<drive>
  <type config:type="symbol">CT_TMPFS</type>
  <partitions config:type="list">
    <partition>
      <mount>/srv</mount>
      <fstopt>size=128M</fstopt>
    </partition>
  </partitions>
</drive>

<drive>
  <type config:type="symbol">CT_TMPFS</type>
  <partitions config:type="list">
    <partition>
      <mount>/var/tmp</mount>
      <fstopt>size=2G</fstopt>
    </partition>
  </partitions>
</drive>
```

## Testing

Unit tests and some manual testing.

### CODE REVIEW

File: package/yast2-storage-ng.changes [hunk 1]
```diff
@@ -1,3 +1,10 @@
+-------------------------------------------------------------------
+Tue Dec 15 15:56:44 UTC 2020 - José Iván López González <jlopez@suse.com>
+
+- AutoYaST: added management of tmpfs file systems (related to
```
Comment (joseivanlopez) [inline package/yast2-storage-ng.changes:4]: NP: actually this is not only related to that jira entry. In fact, such jira is for adding tmpfs support in AutoYaST, although we took the opportunity for resurrecting it in the Partitioner too.

File: test/y2storage/autoinst_proposal_tmpfs_test.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# Copyright (c) [2020] SUSE LLC
```
Comment (joseivanlopez) [inline test/y2storage/autoinst_proposal_tmpfs_test.rb:1]: NP: we add the ruby shebang for *_test.rb* files.

File: test/y2storage/proposal/autoinst_devices_planner_test.rb [hunk 3]
```diff
@@ -327,5 +327,21 @@
...
     end
+
+    context "using CT_TMPFS as type" do
+      let(:partitioning_array) do
+        [{
+          "type" => :CT_TMPFS, "partitions" => [{ "mount" => "/srv" }, { "mount" => "/var/tmp" }]
+        }]
+      end
+
+      it "plans a disk device" do
```
Comment (joseivanlopez) [inline test/y2storage/proposal/autoinst_devices_planner_test.rb:338]: "plans a disk device" ?

**Review by joseivanlopez — APPROVED** [review]
LGTM, only minor things.

**Comment by joseivanlopez** [comment]
BTW, I was checking why the test coverage has decreased, and I see [this line](https://coveralls.io/builds/35732766/source?filename=src%2Flib%2Fy2storage%2Fautoinst_profile%2Fdrive_section.rb#L219) is not tested, which surprised me. Could we add a test for it?

**Comment by imobachgs** [comment]
> BTW, I was checking why the test coverage has decreased, and I see [this line](https://coveralls.io/builds/35732766/source?filename=src%2Flib%2Fy2storage%2Fautoinst_profile%2Fdrive_section.rb#L219) is not tested, which surprised me. Could we add a test for it?

Sure. It surprises me too. I will add a test.

File: test/y2storage/proposal/autoinst_devices_planner_test.rb (hunk 3 above)
Comment (imobachgs) [reply test/y2storage/proposal/autoinst_devices_planner_test.rb]: Oh, `CopyAndPasteError` :grin: Fixed.

**Comment by imobachgs** [comment]
I have updated the PR with a new test. Now, I am considering exporting all `tmpfs` in a single `drive`.

**Review by joseivanlopez — APPROVED** [review]

**Pull request merged**